Awesome Nextflow
=================

A curated list of Nextflow pipelines inspired by other [awesome-*](https://github.com/bayandin/awesome-awesomeness) lists.

Example pipelines
-------------------
* [cbcrg/ampa-nf](https://github.com/cbcrg/ampa-nf) - Automated prediction of protein antimicrobial regions
* [cbcrg/mta-nf](https://github.com/cbcrg/mta-nf) - A method for best alignment of evaluation trees
* [nextflow-io/rnatoy](https://github.com/nextflow-io/rnatoy) - A basic RNA-seq pipeline
* [nextflow-io/rnaseq-nf](https://github.com/nextflow-io/rnaseq-nf) - A basic pipeline for quantification of RNA-seq data using Salmon.

Featured pipelines
-------------------
* [alesssia/YAMP](https://github.com/alesssia/YAMP) - Yet Another Metagenomic Pipeline.
* [cbcrg/piper-nf](https://github.com/cbcrg/piper-nf) - RNA mapping pipeline.
* [cbcrg/kallisto-nf](https://github.com/cbcrg/kallisto-nf) - Nextflow implementation of Kallisto & Sleuth RNA-Seq Tools.
* [cbcrg/unistrap](https://github.com/cbcrg/unistrap) - Reliability measure of inferred phylogenetic trees.
* [CRG-CNAG/CalliNGS-NF](https://github.com/CRG-CNAG/CalliNGS-NF) - Variant Calling Analysis with RNA-Seq data based on GATK best practices.
* [iarcbioinfo/needlestack](https://github.com/IARCbioinfo/needlestack) - Multi-sample somatic variant caller.
* [guigolab/chip-nf](https://github.com/guigolab/chip-nf) - Automated ChIP-seq pipeline.
* [guigolab/grape-nf](https://github.com/guigolab/grape-nf) - Automated RNA-seq pipeline.
* [guigolab/ipsa-nf](https://github.com/guigolab/ipsa-nf) - Integrative Pipeline for Splicing Analyses.
* [h3abionet/h3abionet16S](https://github.com/h3abionet/h3abionet16S) - Integrated pipeline for 16S rDNA diversity analysis.
* [jdidion/atropos](https://github.com/jdidion/atropos/tree/master/paper) - NGS read trimming tool that is specific, sensitive, and speedy.
* [likelet/LncPipe](https://github.com/likelet/LncPipe) - A Nextflow-based pipeline for lncRNA discovery and comprehensive analysis from un processed RNA-seq data.
* [metagenomics/MeRaGENE](https://github.com/metagenomics/MeRaGENE) - Metagenomics rapid gene identification pipeline.
* [mirnylab/distiller-nf](https://github.com/mirnylab/distiller-nf) - Modular Hi-C mapping pipeline.
* [mfoll/robust-regression-caller](https://github.com/mfoll/robust-regression-caller) - Robust regression multi-sample variant caller.
* [montilab/pipeliner](https://github.com/montilab/pipeliner) - Framework for the definition of sequencing data processing workflows.
* [nmdp-bioinformatics/flow](https://github.com/nmdp-bioinformatics/flow) - Consensus assembly and variant calling workflow.
* [robsyme/nf-repeatmasking](https://github.com/robsyme/nf-repeatmasking) - Workflow for automatic repeat detection, classification and masking.
* [sanger-pathogens/companion](https://github.com/sanger-pathogens/companion) - Eukaryotic genome annotation pipeline.
* [SciLifeLab/Sarek](https://github.com/SciLifeLab/Sarek) - Detect germline or somatic variants from normal or tumour/normal WGS.
* [SciLifeLab/NGI-ExoSeq](https://github.com/SciLifeLab/NGI-ExoSeq) - Exome Sequencing Best Practice analysis pipeline.
* [SciLifeLab/NGI-RNAseq](https://github.com/SciLifeLab/NGI-RNAseq) - SciLifeLab NGI RNA-seq analysis pipeline.
* [SciLifeLab/NGI-smRNAseq](https://github.com/SciLifeLab/NGI-smRNAseq) - SciLifeLab NGI small RNA-seq analysis pipeline.
* [SciLifeLab/NGI-MethylSeq](https://github.com/SciLifeLab/NGI-MethylSeq) - SciLifeLab NGI Methylation (Bisulfite-Seq) analysis pipeline.
* [SciLifeLab/NGI-ChIPseq](https://github.com/SciLifeLab/NGI-ChIPseq) - SciLifeLab NGI RNA-seq ChIP-seq analysis pipeline.
* [skptic/concTree](https://github.com/skptic/concTree) - Creates consensus trees using alignment uncertainty from many MSAs.
* [skptic/lncRNA-Annotation-nf](https://github.com/skptic/lncRNA-Annotation-nf) - lncRNA Annotation Pipeline based on STAR, Cufflinks and FEELnc.
* [skptic/paraMSA](https://github.com/skptic/paraMSA/) - Multiple sequence alignments for phylogenetic trees and bootstrap support values.
* [skptic/vast-tools-nf](https://github.com/skptic/vast-tools-nf) - Nextflow Implementation of VAST-TOOLS for profiling alternative splicing events in RNA-Seq data.
* [skptic/tuxedo-nf](https://github.com/skptic/tuxedo-nf) - Nextflow implementation of the Tuxedo Suite of Tools.
* [tburk/smallRNA-meth](https://gitlab.com/tburk/smallRNA-meth) - [Analyzes smallRNA methylation data](https://www.nature.com/articles/nmeth.4610).
* [ZuberLab/mageck-nf](https://github.com/ZuberLab/mageck-nf) - Statistical Analysis of multiplexed CRISPR / shRNA Screens using MAGeCK.

Other pipelines
------------------
* [abreschi/ShortRNA-nf](https://github.com/abreschi/ShortRNA-nf) - Pipeline for processing shortRNA-seq data.
* [andremrsantos/next-pipes-lghm](https://github.com/andremrsantos/next-pipes-lghm) - Eukaryotic genome annotation pipeline.
* [averasd/nextflow-kallisto](https://github.com/AveraSD/nextflow-kallisto)
* [averasd/nextflow-rnastar](https://github.com/AveraSD/nextflow-rnastar)
* [bihealth/ngs_pipelines](https://github.com/bihealth/ngs_pipelines)
* [cbcrg/benchfam](https://github.com/cbcrg/benchfam)
* [cbcrg/grape-nf](https://github.com/cbcrg/grape-nf) - Yet another RNA-Seq pipeline (deprecated)
* [cerebis/meta-sweeper](https://github.com/cerebis/meta-sweeper) - Parametric sweep of simulated microbial communities and metagenomic sequencing.
* [charite/ngspipeline](https://github.com/charite/ngspipeline)
* [chrisquince/DESMAN](https://github.com/chrisquince/DESMAN) - De novo Extraction of Strains from MetAgeNomes.
* [ODiogoSilva/innuca-nf](https://github.com/ODiogoSilva/innuca-nf) - A nextflow implementation of INNUENDO quality control of reads, de novo assembly and contigs quality assessment.
* [fredericlemoine/rna-pipeline](https://github.com/fredericlemoine/rna-pipeline)
* [h3abionet/chipimputation](https://github.com/h3abionet/chipimputation) - Imputation workflow developed as part of the H3 Africa BioNet Hackathon.
* [h3abionet/h3agwas](https://github.com/h3abionet/h3agwas) - GWAS pipeline developed as part of the H3 Africa BioNet Hackathon.
* [HadrienG/pipelines](https://github.com/HadrienG/pipelines)
* [heuermh/bdg-nextflow](https://github.com/heuermh/bdg-nextflow) - Transform BAM files found recursively in a given directory to [ADAM](https://github.com/bigdatagenomics/adam) format.
* [hmkim/workflow](https://github.com/hmkim/workflow/tree/master/nextflow)
* [holtgrewe/ngs_pipelines](https://github.com/holtgrewe/ngs_pipelines) - Nextflow-based pipelines for typical NGS processing tasks.
* [IARCbioinfo/bam_realignment-nf](https://github.com/IARCbioinfo/bam_realignment-nf)
* [IARCbioinfo/bametrics-nf](https://github.com/IARCbioinfo/bametrics-nf) - Compute average metrics from reads that overlap a given set of positions.
* [IARCbioinfo/GVCF_pipeline-nf](https://github.com/IARCbioinfo/GVCF_pipeline-nf)
* [IARCbioinfo/mpileup-nf](https://github.com/IARCbioinfo/mpileup-nf) - Coverage computation with samtools mpileup.
* [IARCbioinfo/mutect-nf](https://github.com/IARCbioinfo/mutect-nf)
* [InSilicoDB/pipeline-kallisto](https://github.com/InSilicoDB/pipeline-kallisto/)
* [InSilicoDB/snp-imputation-nf](https://github.com/InSilicoDB/snp-imputation-nf)
* [jdidion/rna-quick](https://github.com/jdidion/rna-quick) - Rapid analysis of gene and transcript quantification and differential expression.
* [joshua-d-campbell/nf-GATK_Exome_Preprocess](https://github.com/joshua-d-campbell/nf-GATK_Exome_Preprocess) - GATK best practice guide to preprocess whole exome sequencing (WES) data.
* [JWCarlson/interflowscan](https://bitbucket.org/JWCarlson/interflowscan) - Implements the protein analysis components of InterProScan5 using Nextflow.
* [pbelmann/sra-download](https://github.com/pbelmann/sra-download)
* [phelelani/filtering-pipeline](https://github.com/phelelani/filtering-pipeline) - Assembly pipeline for exogenous reads filtered from human RNA-seq data.
* [PlantandFoodResearch/VariantAnalysis](https://github.com/PlantandFoodResearch/VariantAnalysis) - Bioinformatic variant calling pipeline.
* [robsyme/nextflow-annotate](https://github.com/robsyme/nextflow-annotate) - Fungal genome annotation workflow
* [robsyme/nextflow-mapping](https://github.com/robsyme/nextflow-mapping)
* [robsyme/workflows-proteogenomics](https://github.com/robsyme/workflows-proteogenomics) - Versioned, shareable and replicable workflows for proteogenomics.
* [meissnert/nextflow-gtool](https://github.com/meissnert/nextflow-gtool)
* [NationalGenomicsInfrastructure/icing](https://github.com/NationalGenomicsInfrastructure/icing) - HLA typing from OxfordNanopore reads.
* [NBISweden/wgs-structvar](https://github.com/NBISweden/wgs-structvar) - Genome Sequenceing Structural Variation Pipelines.
* [nmdp-bioinformatics/flow-validation](https://github.com/nmdp-bioinformatics/flow-validation)
* [nmdp-bioinformatics/flow-blast-hml](https://github.com/nmdp-bioinformatics/flow-blast-hml)
* [skner/iasi-pipe](https://github.com/skner/iasi-pipe)
* [snewhouse/ngs_nextflow](https://github.com/snewhouse/ngs_nextflow) - Alignment, Variant Calling, GATK Best Practices.
* [wittelab/orchid](https://github.com/Wittelab/orchid) - Novel management, annotation, and machine learning framework for analyzing cancer mutations.

Tutorials
----------
* [CRG course](https://github.com/nextflow-io/crg-course-nov16)
* [Nextflow demos](https://github.com/stevekm/nextflow-demos) - Example Nextflow pipelines and programming techniques
 (by Stephen Kelly)
* [Nextflow tutorial](https://github.com/nextflow-io/hack17-tutorial)
* [H3ABioNet course](https://github.com/shaze/nextflow-course)

Pipelines collections
-------------
* [IARC-nf](https://github.com/IARCbioinfo/IARC-nf) - List of [IARC](https://www.iarc.fr/) bioinformatics nextflow pipelines.
* [NF-Core](https://nf-core.github.io/) - A collection of high quality Nextflow pipelines.

Presentations
-------------
* [](https://rsuchecki.github.io/reproducible/show.html#/) - Reproducbile and Reusable publication and analysis workflow by [bioinforad](https://twitter.com/bioinforad)
* [CIC Show-and-Tell](https://cdn.rawgit.com/robsyme/nextflow-intro/4615e5a/index.html) - Nice Nextflow presentation by [@robsyme](https://github.com/robsyme)

Other links  
---------------
* [Nextflow Hackathon '17](https://github.com/nextflow-io/hack17) - Projects repositorty of the Nextflow Hackathon held in Barcelona, Sept 2017
* [Nextflow Toys](https://github.com/tdelhomme/nextflow_toys) - Scripts for efficiently dealing with Nextflow
* [Nextflow Workbench](http://campagnelab.org/software/nextflow-workbench/) - Workbench that make it easier to write data analysis workflows with Nextflow
* [NF-Core/Cookiecutter](https://github.com/nf-core/cookiecutter) - Template for your own top-end Nextflow pipeline, customised using Cookiecutter.
* [Make to Nextflow](https://github.com/lindenb/xml-patch-make/wiki/Tabix) - Converts Makefile to Nextflow
* [NCBI-Hackathons/SPeW](https://github.com/NCBI-Hackathons/SPeW) - A discussion comparting different workflow frameworks including Nextflow.
* [NGS workflow](https://jmazz.me/blog/NGS-Workflows) Excellent blog post comparing Make/Snakemake/Nextflow by Julian Mazzitelli
